### PR TITLE
Add Delete button to chunk/snippet/plugins-window

### PR DESCRIPTION
**Pull request by achterbahn**
### What does it do?
Add Delete Buttons to the edit windows for chunks/snippets/plugins

### Why is it needed?
For convenience. Now it's possible to delete a chunk/snippet/plugin directly from the edit window. Not only through contextual menu in the tree.

### Related issue(s)/PR(s)
#13211

### CODE REVIEW

**Comment by Jako** [comment]
Nice! A redirect to the manager url would be ok on success.

**Comment by achterbahn** [comment]
I didn't add a button for deleting Templates and TVs. Was not sure if I just can use the delete method on them..

**Comment by pixelchutes** [comment]
Looking good! Yes, if possible this would cover _all_ Elements, so Templates and TVs, too. You should be able to delete just the same, mimicking the current behavior from tree.

**Comment by sottwell** [comment]
What about templates that still have a resource using them? And when deleting a TV, would all of its associated records in the other TV tables be deleted as well?

**Comment by achterbahn** [comment]
Deleting a template where resources are bound to it throws a warning and is not deletable. Deleting a TV removes the TV and all the entries in the table site_tmplvar_contentvalues. 
The delete buttons should reflect this behaviour.

**Comment by sottwell** [comment]
site_tmplvar_templates and site_tmplvar_access?

**Comment by Jako** [comment]
@sottwell He uses the default remove processors, so this is covered

https://github.com/achterbahn/revolution/blob/c43a25483b0e39a0a6e8fa700f7f96ecc33475d1/manager/assets/modext/sections/element/chunk/update.js#L77-L80

File: manager/assets/modext/sections/element/chunk/update.js [hunk 1]
```diff
@@ -64,5 +70,21 @@ Ext.extend(MODx.page.UpdateChunk,MODx.Component, {
...
+				text: _('chunk_delete_confirm')
+				,url: MODx.config.connector_url
+				,params: {
+						action: 'element/chunk/remove'
+						,id: this.record.id
+				}
+				,listeners: {
+						success: {
+							fn:function(r) {
+								MODx.loadPage('?'); //not sure about this
```
Comment (Jako) [inline manager/assets/modext/sections/element/chunk/update.js]: Try `MODx.loadPage(MODx.config.manager_url);`

Comment (achterbahn) [inline manager/assets/modext/sections/element/chunk/update.js]: Thanks.

**Comment by achterbahn** [comment]
The remove processor (/core/model/modx/processors/element/template/remove.class.php) of a template checks if a resource is using this template.

In contrast the remove processor of TVs is not checking. I added a check if a TV is used by a template or a resource.
`      public function beforeRemove() {

      /* get tv relational tables */
      $this->TemplateVarTemplates = $this->object->getMany('TemplateVarTemplates');
      $this->TemplateVarResources = $this->object->getMany('TemplateVarResources');
      $this->TemplateVarResourceGroups = $this->object->getMany('TemplateVarResourceGroups');

        /* check if any template uses this TV */
        $tvts = $this->object->getMany('TemplateVarTemplates',array(
          'tmplvarid' => $this->object->get('id')
        ));
        if (count($tvts) > 0) {
          foreach ($tvts as $tvt) {
            $ms .= "<br />\n Template ID: ".$tvt->get('templateid');
          }
        }

        if ($ms != "") {
          return $this->modx->lexicon('tv_inuse_template').$ms;
        } else {
          return true;
        }

    }
`

**Comment by OptimusCrime** [comment]
Awesome PR, just a few things about the code style:

- PHP code is indented with two spaces, the rest of the project uses four spaces
- The JS code is indented with tabs, the rest of the project uses four spaces

**Comment by achterbahn** [comment]
you're absolutely right @OptimusCrime. I'm glad I could make the clone/commit/push things in git. Next step: working on the cosmetics :-) 
There should be something like a MODX Contribute workshop from time to time. But I hope I get more fluent with all the rules and stuff.
Thanks for your input.

**Comment by Jako** [comment]
Better check if that processor is used somewhere else before changing it:
core/model/modx/processors/element/tv/remove.class.php

I think it will not be a BC, but it could.

**Comment by Jako** [comment]
Merged into 2.x

**Pull request closed without merging**